web/src/main/java/org/cbioportal/web/config/DataAccessTokenControllerConfig.java

```diff
@@ -0,0 +1,61 @@
+/*
+* Copyright (c) 2020 The Hyve B.V.
+*
+* This library is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS
+* FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+* is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+* obligations to provide maintenance, support, updates, enhancements or
+* modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+* liable to any party for direct, indirect, special, incidental or
+* consequential damages, including lost profits, arising out of the use of this
+* software and its documentation, even if Memorial Sloan-Kettering Cancer
+* Center has been advised of the possibility of such damage.
+*/
+
+/*
+* This file is part of cBioPortal.
+*
+* cBioPortal is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU Affero General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+package org.cbioportal.web.config;
+
+import org.cbioportal.web.DataAccessTokenController;
+import org.cbioportal.web.OAuth2DataAccessTokenController;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.context.annotation.PropertySources;
+
+@Configuration
+@PropertySources({
```

Comment:
I think this needs some tweaking.  Since I deploy war with portal.properties rolled in, to avoid a FileNotFoundException, I had to do the following:

> @PropertySource(value="file:///${portal_home}/portal.properties", ignoreResourceNotFound=true)

I imagine you want to look in war first and if not found, then environment?  Maybe you need to place ignoreResourceNotFound on both @PropertySource and then do something smart in the condition class?

Reply:
I have updated to this:
```
@Configuration
@PropertySources({
    @PropertySource("file:///${PORTAL_HOME}/portal.properties", ignoreResourceNotFound=true),
    @PropertySource("classpath:portal.properties", ignoreResourceNotFound=true)
})
```

I do not think the Condition classes need to be updated since these check for presence of the `dat.method` prop. If not present, the beans are not created without making a fuss.